From ce3d8a99a2e5c5ef1ff0faf5c290a113119be582 Mon Sep 17 00:00:00 2001
From: David Ferruz <dferruz@maxlinear.com>
Date: Fri, 5 Jul 2024 07:18:22 +0200
Subject: [PATCH] WLANRTSYS-82831 [Octopus][WAV700][B0][11BE][Regression]:
 Observed very low EHT Throughput or Traffic stuck for one of the STA while
 running UDP DL Traffic with 2 2x2 STA's when AP is configured to 6G320MHz

Problem:
Observed xput drops and Intel BE200 STAs stop responding to RTS for
several seconds
Also, fwlog stops (WLANRTSYS-82833 [Octopus][WAV700][B0][Regression]: Observed FW Logs stuck issue after some time of capturing FW Logs)

Solution:
Detected the issue is happening since WLANRTSYS-82491 where merged in
CV2073, thus reverting the changes until they will be debugged to
unblock master branch

Revert "WLANRTSYS-82491 [WAV700][MLD] Beacon protection is not enabled by default in Latest builds."

This reverts commit 9ed43f02373afae2c0df5f938f17b4e23fbba5f1.

Revert "WLANRTSYS-82491 [WAV700][MLD] Beacon protection is not enabled by default in Latest builds."

This reverts commit b03d74dae98262fbad9aea89e2803116ea97ed37.
---
 hostapd/mxl_config.c | 12 ------------
 src/ap/ap_config.c   |  6 ------
 2 files changed, 18 deletions(-)

diff --git a/hostapd/mxl_config.c b/hostapd/mxl_config.c
index 6a30951b3..aaf7b0b85 100644
--- a/hostapd/mxl_config.c
+++ b/hostapd/mxl_config.c
@@ -2054,18 +2054,6 @@ int mxl_hostapd_config_check(struct hostapd_config *conf, int full_config)
 		}
 	}
 #endif /* CONFIG_IEEE80211BE_MXL_MLO */
-#ifdef CONFIG_IEEE80211BE
-	if (full_config && conf->ieee80211be) {
-		int i;
-		for (i = 0; i < conf->num_bss; i++) {
-			if (!conf->bss[i]->disable_11be) {
-				if ((conf->bss[i]->wpa_key_mgmt & WPA_KEY_MGMT_OWE) || wpa_key_mgmt_sae(conf->bss[i]->wpa_key_mgmt)) {
-				    conf->bss[i]->rsn_pairwise |= WPA_CIPHER_GCMP_256;
-				}
-			}
-		}
-	}
-#endif /* CONFIG_IEEE80211BE */
 
 	return 0;
 }
diff --git a/src/ap/ap_config.c b/src/ap/ap_config.c
index ecdcd7c9b..aca50cb28 100644
--- a/src/ap/ap_config.c
+++ b/src/ap/ap_config.c
@@ -1621,12 +1621,6 @@ static int hostapd_config_check_bss(struct hostapd_bss_config *bss,
 		wpa_printf(MSG_INFO,
 			   "Disabling IEEE 802.11be as IEEE 802.11ax is disabled for this BSS");
 	}
-
-	if (full_config && conf->ieee80211be && !bss->disable_11be && !bss->beacon_prot) {
-		bss->beacon_prot = 1;
-		wpa_printf(MSG_INFO,
-			   "Enabling beacon protection as IEEE 802.11be is enabled for this BSS");
-	}
 #endif /* CONFIG_IEEE80211BE */
 
 	if (full_config && bss->ignore_broadcast_ssid && conf->mbssid) {
-- 
2.43.0

